File: Models/Inventory.cs
using Microsoft.AspNetCore.Mvc.ModelBinding;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace SeedInventory.Models
{
  public class Inventory
  {
    public int Id { get; set; }
    public string Name { get; set; }
    public int Kernels { get; set; }
    public int KernelsUsed { get; set; }
    public DateTime DateCreated { get; set; }
    public DateTime DateUpdated { get; set; }
    public List<InventoryRequest> InventoryRequests { get; set; }
    public Inventory GetCopy(Inventory obj)
    {
      var item = new Inventory()
      {
        Id = obj.Id,

        Name = obj.Name,
        Kernels = obj.Kernels
      };

      return item;
    }
  }

  public class InventoryRequest
  {
    public int Id { get; set; }
    public int InventoryId { get; set; }
    public int RequestedKernels { get; set; }
    public bool Approved { get; set; } = false;
    public bool Locked { get; set; } = false;
    public DateTime DateCreated { get; set; }
    public DateTime DateUpdated { get; set; }
    [ForeignKey("InventoryId")]
    public virtual Inventory Inventory { get; set; }
  }

  public class InventoryData
  {
    public List<Inventory> Inventory { get; set; }
    [JsonProperty("requests")]
    public List<InventoryRequest> InventoryRequests { get; set; } 
  }

  public class ApiResponse<T>
  {
    public bool Status { get; set; }
    public T Item { get; set; }
    public ModelStateDictionary ModelState { get; set; }
    public string Message { get; set; }
  }
}

File: src/app/inventory.ts
export class Inventory {
  Id: number;
  Name: string;
  Kernels: number;
  KernelsUsed: number;
  KernelsAvailable: number;
  DateCreated?: Date;
  DateUpdated?: Date;
  InventoryRequests?: InventoryRequestSelectable[];
  constructor(item: Inventory) {
    this.Id = item.Id;
    this.Name = item.Name;
    this.Kernels = item.Kernels;
    this.KernelsUsed = item.KernelsUsed;
    this.KernelsAvailable = item.KernelsAvailable;
    this.InventoryRequests = [];
    item.InventoryRequests.forEach(i => this.InventoryRequests.push(new InventoryRequestSelectable(i)));
  }
};

export class InventorySelectable extends Inventory {
  Selected: boolean;
  Order?: number;
  Included: boolean;
};

export class InventoryRequest {
  Id: number;
  InventoryId: number;
  RequestedKernels: number;
  Approved: boolean;
  Locked: boolean;
  DateCreated?: Date;
  DateUpdated?: Date;
  Inventory?: Inventory;
  constructor(item: InventoryRequest) {
    this.Id = item.Id;
    this.InventoryId = item.InventoryId;
    this.RequestedKernels = item.RequestedKernels;
    this.Approved = item.Approved;
    this.Locked = item.Locked;
    if (item.Inventory) {
      this.Inventory = {
        Id: item.Inventory.Id,
        Name: item.Inventory.Name,
        Kernels: item.Inventory.Kernels,
        KernelsUsed: item.Inventory.KernelsUsed,
        KernelsAvailable: item.Inventory.KernelsAvailable,
        DateCreated: item.Inventory.DateCreated,
        DateUpdated: item.Inventory.DateUpdated
      }
    }
  };
};

export class InventoryRequestSelectable extends InventoryRequest {
  Selected: boolean;
  Order?: number;
  Included: boolean;
};


